### PR TITLE
EOS-18337: Docs: Update testing documentation and docker deployment guide

### PR DESCRIPTION
**Pull request by dmitrydb**
I prepared doc with instruction how to deploy docker and configure testing environment. Also I updated the test scenario for EOS-18337

-----
[View rendered docs/testing.md](https://github.com/Seagate/cortx-prvsnr/blob/EOS-18337-docs/docs/testing.md)

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+
+## Installation
+
+#### Docker configuration
+
+The base Docker installation methods for popular Linux distributions are listed here:
+	https://docs.docker.com/engine/install/
+
+Docker is not supported by **Fedora 33** officially. On my host system I used the following method (Moby based approach):
```
Comment (andkononykhin2) [inline docs/testing.md:18]: please separate that into a sub-part with header like `Docker on Fedora ...`

File: docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+1. Delete the old version of docker (if it was installed)
```
Comment (andkononykhin2) [inline docs/testing.md:20]: Can we fold that using e.g. collapsive section (like described [here](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab))

File: docs/testing.md (hunk 1 above)
Comment (andkononykhin2) [inline docs/testing.md:18]: Could you please avoid 1st person personal pronouns?

File: docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    User root
+    UserKnownHostsFile /dev/null
+    StrictHostKeyChecking no
+    IdentityFile /tmp/pytest-of-fdmitry/pytest-10/id_rsa.test
+    IdentitiesOnly yes
+    LogLevel FATAL
+  
+  ```
+
+- Install provisioner API to created docker container
```
Comment (andkononykhin2) [inline docs/testing.md:118]: It's not only API. Better to say: You may then setup a provisioner inside the container using the `local` source mode.

And add marks:
TBD Test other types of sources.
TBD Update with examples for multi node setups with HA mode enabled.

File: docs/testing.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+- Install provisioner API to `venv`: 
+
+  ```bash
+  pip install -e api/python[test]
+  ```
+
+- Start the docker environment creation
+
+  ```bash
+  pytest test/build_helpers -k test_build_setup_env -s --root-passwd root --nodes-num 1
```
Comment (andkononykhin2) [inline docs/testing.md:91]: Could you please also make a note that it's possible to provide custom command line args using. Makes sense to copy-paste that part from #926 description.

**Review by andkononykhin2 — CHANGES_REQUESTED** [review]
@dmitrydb Thanks. It's a very helpful doc for anyone who wants to start trying integration testing based on docker. I marked some places where we need to adjust the content though.

File: docs/testing.md (hunk 4 above)
Comment (dmitrydb) [reply docs/testing.md:114]: Done

File: docs/testing.md (hunk 3 above)
Comment (dmitrydb) [reply docs/testing.md]: Done

File: docs/testing.md (hunk 1 above)
Comment (dmitrydb) [reply docs/testing.md]: I used person pronouns to underline that it was just a mine experience. Done.

File: docs/testing.md (hunk 2 above)
Comment (dmitrydb) [reply docs/testing.md]: Done

File: docs/testing.md (hunk 1 above)
Comment (dmitrydb) [reply docs/testing.md]: Done

**Review by andkononykhin2 — APPROVED** [review]

**Review by ypise — APPROVED** [review]

**Pull request merged**